libcodechecker/analyze/analysis_manager.py

```diff
@@ -157,6 +157,20 @@ def check(check_data):
                           action.analyzer_type, source_file_name))
             else:
                 # Analysis failed.
+                failed_dir = os.path.join(output_dir, "failed")
+                if not os.path.exists(failed_dir):
+                    os.makedirs(failed_dir)
+                err_file = os.path.basename(
+                    # Remove .plist extension
+                    rh.analyzer_result_file[:len(rh.analyzer_result_file)-6])
```

Comment:
There is a python function `os.path.splitext` which can split the extension from a filename. I would recommend using that.